src/http/input/metadata/LinkRelParser.ts

```diff
@@ -25,23 +25,13 @@ export class LinkRelParser extends MetadataParser {
   public async handle(input: { request: HttpRequest; metadata: RepresentationMetadata }): Promise<void> {
     const link = input.request.headers.link ?? [];
     const entries: string[] = Array.isArray(link) ? link : [ link ];
-    for (const entry of entries) {
-      this.parseLink(entry, input.metadata);
-    }
+    this.parseLink(entries, input.metadata);
   }
 
-  protected parseLink(linkEntry: string, metadata: RepresentationMetadata): void {
-    const { result, replacements } = transformQuotedStrings(linkEntry);
-    for (const part of splitAndClean(result)) {
-      const [ link, ...parameters ] = part.split(/\s*;\s*/u);
-      if (/^[^<]|[^>]$/u.test(link)) {
-        this.logger.warn(`Invalid link header ${part}.`);
-        continue;
-      }
-      for (const { name, value } of parseParameters(parameters, replacements)) {
-        if (name === 'rel' && this.linkRelMap[value]) {
-          metadata.add(this.linkRelMap[value], namedNode(link.slice(1, -1)));
-        }
+  protected parseLink(linkHeaders: string[], metadata: RepresentationMetadata): void {
+    for (const { link, params } of parseLinkHeader(linkHeaders)) {
+      if (params.rel && this.linkRelMap[params.rel]) {
```

Comment:
RFC 8288 says `The rel parameter MUST be present` so we can make this slightly easier.